ships/0043-buildrun-executor-field.md

```diff
@@ -0,0 +1,189 @@
+
+## Motivation
+
+This enhancement is motivated by the need to support PipelineRun as an alternative execution mode to TaskRun in Shipwright's build logic. As part of refactoring Shipwright to execute builds in Tekton PipelineRuns, cluster admins will be able to switch between the object type Shipwright creates to run builds. This is a necessary prerequisite for fanning out parallel multi-arch builds.
+
+The introduction of PipelineRun support as an "alpha" feature (disabled by default) ensures we don't break existing TaskRun-based behavior while allowing experimentation with the PipelineRun "mode" for operating builds. However, with multiple execution modes available, users and operators need visibility into which specific Tekton resource (TaskRun or PipelineRun) is actually executing their BuildRun to enable proper debugging, monitoring, and troubleshooting.
+
+### Goals
+
+- Provide clear visibility into which Tekton resource (TaskRun or PipelineRun) is executing a BuildRun
+- Enable better debugging and troubleshooting capabilities for build execution issues
+- Support monitoring and observability tools that need to correlate BuildRun status with underlying Tekton resources
+- Maintain backward compatibility with existing BuildRun resources
+- Replace the deprecated `TaskRunName` field with a more comprehensive `Executor` field that supports both TaskRun and PipelineRun execution modes
+
+### Non-Goals
+
+- Modifying the core build execution logic
+- Changing how BuildRuns are created or managed
+- Adding executor information to Build resources (only BuildRun status)
+
+## Proposal
+
+### User Stories
+
+#### Story 1: Introduce Executor field for runner kinds 
+As a developer, I want a new Executor field to be added to the BuildRunStatus to define the resource responsible for executing a build.
+
+### Implementation Notes
+
+The new `executor` field will be added to the `BuildRunStatus` struct with the following structure:
+
+```go
+type BuildRunStatus struct {
+    // ... existing fields ...
+    
+    // TaskRunName is the name of the TaskRun responsible for executing this BuildRun.
+    //
+    // Deprecated: Use Executor instead to describe the taskrun.
+    // +optional
+    TaskRunName *string `json:"taskRunName,omitempty"`
+
+    // Executor is the name and kind of the resource responsible for executing this BuildRun.
+    //
+    // +optional
+    Executor *BuildExecutor `json:"executor,omitempty"`
+    
+    // ... other existing fields ...
+}
+
+// BuildExecutor defines the name and kind of the build runner.
+type BuildExecutor struct {
+    // Name is the name of the TaskRun or PipelineRun that was created to execute this BuildRun
+    Name string `json:"name"`
+    
+    // Kind is the kind of the object that was created to execute the BuildRun (e.g., "TaskRun", "PipelineRun")
+    Kind string `json:"kind"`
```

Comment:
If we ever support a non-Tekton executor, then we should also have a `group` field:


```suggestion
    // Group is the API group of the object that was created to execute the BuildRun (e.g. "tekton.dev")
    Group string `json:"group"`
    
    // Kind is the kind of the object that was created to execute the BuildRun (e.g., "TaskRun", "PipelineRun")
    Kind string `json:"kind"`
```